diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/StringTemplateFormatRule.kt

```diff
@@ -77,4 +85,20 @@ class StringTemplateFormatRule(private val configRules: List<RulesConfig>) : Rul
             }
         }
     }
+
+    @Suppress("UnsafeCallOnNullableType")
+    private fun bracesCanBeOmitted(node: ASTNode): Boolean {
+        val onlyOneRefExpr = node
+                .findAllNodesWithSpecificType(REFERENCE_EXPRESSION)
+                .singleOrNull()
+                ?.treeParent
+                ?.elementType == LONG_STRING_TEMPLATE_ENTRY
+        return if (onlyOneRefExpr) {
+            !(node.treeNext.text.first().isLetterOrDigit() // checking if first letter is valid
+                    || node.treeNext.text.startsWith("_"))
+                    || node.treeNext.elementType == CLOSING_QUOTE
+        } else {
+            node.hasAnyChildOfTypes(FLOAT_CONSTANT, INTEGER_CONSTANT) // it also fixes "${1.0}asd" cases
```

Comment:
Will it be corrected into `$1.0asd` then? But it's not correct too - for literals there will be just implicit `toString` call, and if we remove braces, we need to remove `$` as well.

Reply:
Oh, I forgot that we already are able to remove `$` as well.